x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/utils/indices.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+
+import { i18n } from '@kbn/i18n';
+
+import { SyncStatus, ConnectorStatus } from '../../../../common/types/connectors';
+import { ElasticsearchIndexWithIngestion } from '../../../../common/types/indices';
+
+import { ElasticsearchViewIndex, IngestionMethod, IngestionStatus } from '../types';
+
+export function getIngestionMethod(index?: ElasticsearchIndexWithIngestion): IngestionMethod {
```

Comment:
I'm not sure if this is better than doing a typescript check and checking typeof everywhere. Seems like that could be more robust.

Reply:
Can't do that, typeof only returns JavaScript types, not TypeScript ones (because TypeScript is a compile-time language, not a runtime one). We *could* write some typeguards a la isConnector/isCrawler/isAPI 🤔

Reply:
Yeah that's what I meant (forgot the typescript word for it haha), doesn't have to be a part of this PR but I think we should be doing type checks for situations like this rather than using an enum as a proxy.